pkg/registry/apis/provisioning/jobs/persistentstore.go

```diff
@@ -379,12 +379,23 @@ func (s *persistentStore) cleanupClaims(ctx context.Context) error {
 // The job name is not honoured. It will be overwritten with a name that fits the job.
 //
 // This saves it if it is a new job, or fails with `apierrors.IsAlreadyExists(err) == true` if one already exists.
+// If one already exists, it is read and returned.
 func (s *persistentStore) Insert(ctx context.Context, job *provisioning.Job) (*provisioning.Job, error) {
 	s.generateJobName(job) // Side-effect: updates the job's name.
 
 	obj, err := s.jobStore.Create(ctx, job, nil, &metav1.CreateOptions{})
 	if apierrors.IsAlreadyExists(err) {
-		return nil, apifmt.Errorf("job '%s' in '%s' already exists: %w", job.GetName(), job.GetNamespace(), err)
+		obj, err = s.jobStore.Get(ctx, job.GetName(), &metav1.GetOptions{})
+		if err != nil {
+			return nil, apifmt.Errorf("job '%s' in '%s' already exists but failed to get it: %w", job.GetName(), job.GetNamespace(), err)
+		}
+
+		existing, ok := obj.(*provisioning.Job)
+		if !ok {
+			return nil, apifmt.Errorf("unexpected object type %T", obj)
+		}
+
+		return existing, apifmt.Errorf("job '%s' in '%s' already exists: %w", job.GetName(), job.GetNamespace(), err)
```

Comment:
It's unusual to return the existing object here when things error given go pattern of if error returned ignored. 

Besides that, at this stage also the error returned up is not actually the `apierrors.IsAlreadyExists`, is it? I think it gets replaced by the 388 was the get error. 


In any case, I think it depends on what we want to return in our APIs and logging. In my opinion, it's fine to return the 409 in this situation and log that it already exists in different spots.